cmd/gardenadm/app/app.go

```diff
@@ -5,35 +5,61 @@
 package app
 
 import (
+	"fmt"
 	"os"
 
 	"github.com/spf13/cobra"
 	"k8s.io/cli-runtime/pkg/genericiooptions"
+	"k8s.io/klog/v2"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	logzap "sigs.k8s.io/controller-runtime/pkg/log/zap"
 
+	"github.com/gardener/gardener/pkg/gardenadm/cmd"
 	"github.com/gardener/gardener/pkg/gardenadm/cmd/bootstrap"
 	"github.com/gardener/gardener/pkg/gardenadm/cmd/connect"
 	"github.com/gardener/gardener/pkg/gardenadm/cmd/discover"
 	initcmd "github.com/gardener/gardener/pkg/gardenadm/cmd/init"
 	"github.com/gardener/gardener/pkg/gardenadm/cmd/join"
 	"github.com/gardener/gardener/pkg/gardenadm/cmd/token"
 	"github.com/gardener/gardener/pkg/gardenadm/cmd/version"
+	"github.com/gardener/gardener/pkg/logger"
 )
 
 // Name is a const for the name of this component.
 const Name = "gardenadm"
 
 // NewCommand creates a new cobra.Command for running gardenadm.
 func NewCommand() *cobra.Command {
-	opts := &Options{
+	opts := &cmd.Options{
 		IOStreams: genericiooptions.IOStreams{In: os.Stdin, Out: os.Stdout, ErrOut: os.Stderr},
 	}
 
 	cmd := &cobra.Command{
 		Use:   Name,
 		Short: Name + " bootstraps and manages autonomous shoot clusters in the Gardener project.",
+		PersistentPreRunE: func(cmd *cobra.Command, _ []string) error {
+			if err := opts.Complete(); err != nil {
+				return err
+			}
+
+			if err := opts.Validate(); err != nil {
+				return err
+			}
+
+			var err error
+			opts.Log, err = logger.NewZapLogger(opts.LogLevel, opts.LogFormat, logzap.WriteTo(opts.IOStreams.Out))
```

Comment:
Can you move the logger initialization to `opts.Complete`?

Reply:
I had done it this way, but due to https://github.com/gardener/gardener/pull/11390#discussion_r1955856366, I moved it out (otherwise, we loose our validation/the logger instantiation fails when invalid log levels/formats are provided).